src/zeek.bif

```diff
@@ -2129,10 +2129,31 @@ function global_ids%(%): id_table
 		if ( id->HasVal() )
 			rec->Assign(7, id->GetVal());
 
+		return rec;
+	};
+
+	auto ids = zeek::make_intrusive<zeek::TableVal>(id_table);
+
+	for ( const auto& [_, id] : zeek::detail::global_scope()->Vars() )
+		{
 		auto id_name = zeek::make_intrusive<zeek::StringVal>(id->Name());
+		auto rec = to_script_id_rec(id);
 		ids->Assign(std::move(id_name), std::move(rec));
 		}
 
+	// We do not track modules as identifiers, ad-hoc instantiate one
+	// that is re-used here.
+	auto mod_id = zeek::make_intrusive<zeek::detail::ID>("<<void>>", zeek::detail::SCOPE_GLOBAL, false /* is_export */);
+	mod_id->SetType(zeek::base_type(zeek::TYPE_VOID));
+	auto base_rec = to_script_id_rec(mod_id);
+	base_rec->Assign(0, "module");  // Assign type_name as "module"
```

Comment:
Seems feels a bit indirect. I get the desire to factor out the logic into the lambda, but I think it would be more readable to just initialize module-specific records here directly, then you'd not need the fake mod_id.